lib/auth/bot.go

```diff
@@ -168,14 +174,86 @@ func (a *Server) validateGenerationLabel(ctx context.Context, username string, c
 	return nil
 }
 
+func (a *ServerWithRoles) updateBotAuthentications(ctx context.Context, req *certRequest) error {
```

Comment:
I assume we need to account for there not being an ident.BotInstanceID if this is a cert from pre-BotInstances that is renewing.

Reply:
Yep, good call, I've added a case to handle that.